presto-main/src/main/java/com/facebook/presto/operator/project/InterpretedCursorProcessor.java

```diff
@@ -92,27 +94,29 @@ private static ExpressionInterpreter getExpressionInterpreter(
     }
 
     @Override
-    public int process(ConnectorSession session, RecordCursor cursor, int count, PageBuilder pageBuilder)
+    public int process(ConnectorSession session, DriverYieldSignal yieldSignal, RecordCursor cursor, int count, PageBuilder pageBuilder)
     {
         checkArgument(!pageBuilder.isFull(), "page builder can't be full");
         checkArgument(count > 0, "count must be > 0");
+        requireNonNull(yieldSignal, "yieldSignal is null");
 
-        int position = 0;
+        // make sure we at least process one position
+        if (pageBuilder.isFull() || !cursor.advanceNextPosition()) {
+            return 0;
+        }
+        project(cursor, pageBuilder);
```

Comment:
1. Is this what we want to do here? I would talk with @dain 
2. If we decide we are ok with this, we should rewrite the for-loop into something else so that we don't duplicate the loop body. I'm good with `while (true)` loops if that necessary.